verification/spec/connection-handshake/ch.tla

```diff
@@ -0,0 +1,188 @@
+    - counterpartyClientID -- a client identifier
+      Stores the counterparty client identifier associated with this connection end.
+ ***************************************************************************)
+ConnectionEnd ==
+    [
+        state : ConnectionStates,
+        connectionID : ConnectionIDs \union {Null},
+        clientID : ClientIDs \union {Null},
+        counterpartyConnectionID : ConnectionIDs \union {Null},
+        counterpartyClientID : ClientIDs \union {Null}
+    ]      
+
+(********************************** Helper functions
+ ***************************************************************************)
+
+\* Constructs a ConnectionEnd object for the given party in state UNINIT.
+getConnectioEnd(targetParty) ==
+    IF targetParty = "A"
+    THEN 
+        [   state |-> "UNINIT",
+            connectionID |-> "connAtoB",
+            clientID |-> "clientA",
+            counterpartyConnectionID |-> "connBtoA",
+            counterpartyClientID |-> "clientB" ]
+    ELSE IF targetParty = "B"
+         THEN
+            [   state |-> "UNINIT",
+                connectionID |-> "connBtoA",
+                clientID |-> "clientB",
+                counterpartyConnectionID |-> "connAtoB",
+                counterpartyClientID |-> "clientA" ]
+         ELSE Null    
+
+
+\* Constructs a 'Init' Datagram.
+\* Fields are hardcoded for the moment
+getInitDatagram(targetParty) ==
+    [ type |-> "ConnOpenInit", 
+      connectionEnd |-> getConnectioEnd(targetParty) ]
+
+
+isPartyInState(p, state) ==
+    parties[p].connectionEnd.state = state
+
+
+\* Would be ideal to import the following function.
+\* Cf. ibc-rs#55 (file: ConnectionHandlers.tla)
+getConnectionID(p) ==
+    IF p = "A"
+    THEN "connAtoB"
+    ELSE IF p = "B"
+         THEN "connBtoA"
+         ELSE Null  
+
+
+(********************************** Environment functions
+ ***************************************************************************)
+
+    
+\* The environment generates the ConnOpenInit datagram, sending it to 'party'
+GenerateConnOpenInit(targetParty) ==
+    LET initDatagram == getInitDatagram(targetParty)
+    IN inboundDatagrams' = [
+                                inboundDatagrams EXCEPT
+                                ![targetParty] = inboundDatagrams[targetParty] \union { initDatagram } 
+                           ]
+    /\ UNCHANGED <<parties>>
+
+
+EnvStep ==
+    \E targetParty \in Parties :
+        GenerateConnOpenInit(targetParty)
+
+
+(**************************** Party functions
+ ***************************************************************************)
+
+ConnectionEndInit ==
+    [state |-> "UNINIT",
+     connectionID |-> Null,
+     clientID |-> Null,
+     counterpartyConnectionID |-> Null,
+     counterpartyClientID |-> Null ] 
+
+
+(**
+ * Connection Handshake Module handlers
+ **)
+
+CHModuleHandleInit(party) ==
+    /\ isPartyInState(party, "UNINIT")
+    /\  LET connOpenInitDgrs == {dgr \in inboundDatagrams[party] : 
```

Comment:
You can simplify this. Handler should just consume an inMsg (see for example how we do it in fastSync spec), and you directly access it inside the handler.